### PR TITLE
feat: add support for device preferences commands

### PR DESCRIPTION
**Pull request by rossiam**
* added support for list/get, create and update for device preferences (delete is unavailable at this time)
* added a couple of useful functions wrapping common inquirer functions (lots more could be done with this--I just did what I needed for this specific functionality)
* fixed a couple of incorrect command descriptions I noticed
* updated @smarthings/core-sdk version
* updated inquirer version in 

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **[CONTRIBUTING](../CONTRIBUTING.md)** document
- [x] My code follows the code style of this project (`lerna run lint` produces no warnings/errors)
- [x] Any required documentation has been added
- [x] I have added tests to cover my changes

### CODE REVIEW

File: packages/cli/src/commands/devicepreferences/create.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+		await super.setup(args, argv, flags)
+
+		await inputAndOutputItem(this, { tableFieldDefinitions },
+			(_, input: DevicePreferenceCreate) => this.client.devicePreferences.create(input),
+			userInputProcessor(this))
+	}
+
+	async getInputFromUser(): Promise<DevicePreferenceCreate> {
+		const name = await askForString('Preference name:',
+			input => !input || input.match(/^[a-z][a-zA-Z0-9]{2,23}$/) ? true : 'must be camel case starting with a lowercase letter and 3-24 characters')
```
Comment (john-u) [inline packages/cli/src/commands/devicepreferences/create.ts:38]: We could write `must be camelCase. . .` in the prompt to be extra clear.

File: packages/cli/src/commands/devicepreferences/create.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+				choices: ['text', 'password', 'paragraph'],
+				default: 'text',
+			})).stringType as 'text' | 'password' | 'paragraph'
+			const defaultValue = await askForString('Optional default value.',
+				input => {
+					if (minLength !== undefined && input.length < minLength) {
+						return `default must be at least minLength (${minLength}) characters`
+					}
+					if (maxLength !== undefined && input.length > maxLength) {
+						return `default must be no more than maxLength (${maxLength}) characters`
```
Comment (john-u) [inline packages/cli/src/commands/devicepreferences/create.ts:118]: We should make these (and the similar messaging in `user-query.ts`) consistent and both some form of `no more/no less than` or `at most/at least`. I don't have strong feelings either way.

File: packages/cli/src/commands/devicepreferences/create.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+		if (preferenceType === 'enumeration') {
+			const firstName = await askForRequiredString('Enter a name for the first option.')
+			let value = await askForRequiredString('Enter a value for the first option.')
+
+			const options: { [name: string]: string } = { [firstName]: value }
+
+			let name: string | undefined
+			do {
+				name = await askForString('Enter a name for the next option, if any.')
+				if (name) {
```
Comment (john-u) [inline packages/cli/src/commands/devicepreferences/create.ts:141]: Is it clear to the user here that entering nothing is how they continue? If not, we should add some text stating so.

File: packages/lib/src/__tests__/user-query.test.ts [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+			expect(validateFunction('a')).toBe(true)
+		})
+	})
+
+	describe('askForInteger', () => {
+		it.each`input   | expected
+				${'-1'} | ${-1}
+				${'0'}  | ${0}
+				${'7'}  | ${7}
+				${''}   | ${undefined}`('returns number entered', async ({ input, expected }) => {
```
Comment (john-u) [inline packages/lib/src/__tests__/user-query.test.ts:83]: The formatting is a little off here.

File: packages/lib/src/__tests__/user-query.test.ts [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+			expect(validateFunction('1')).toBe('must be no more than 0')
+		})
+	})
+
+	describe('askForNumber', () => {
+		it.each`input   | expected
+				${'-1'} | ${-1}
+				${'0'}  | ${0}
+				${'3.141592'}  | ${3.141592}
+				${''}   | ${undefined}`('returns number entered', async ({ input, expected }) => {
```
Comment (john-u) [inline packages/lib/src/__tests__/user-query.test.ts:168]: Formatting is off here as well.

File: packages/cli/src/commands/devicepreferences/create.ts (hunk 1 above)
Comment (rossiam) [reply packages/cli/src/commands/devicepreferences/create.ts]: Fixed.

File: packages/cli/src/commands/devicepreferences/create.ts (hunk 2 above)
Comment (rossiam) [reply packages/cli/src/commands/devicepreferences/create.ts:118]: changed

File: packages/cli/src/commands/devicepreferences/create.ts (hunk 3 above)
Comment (rossiam) [reply packages/cli/src/commands/devicepreferences/create.ts:141]: I was thinking ", if any" was enough but I've changed ", if any" to " or press enter to continue" to make it more clear.

File: packages/lib/src/__tests__/user-query.test.ts (hunk 4 above)
Comment (rossiam) [reply packages/lib/src/__tests__/user-query.test.ts:83]: I think in this case, that is just an artifact of github using 8 spaces for a tab.

File: packages/lib/src/__tests__/user-query.test.ts (hunk 5 above)
Comment (rossiam) [reply packages/lib/src/__tests__/user-query.test.ts]: Fixed the formatting here.

File: packages/lib/src/__tests__/user-query.test.ts (hunk 4 above)
Comment (john-u) [reply packages/lib/src/__tests__/user-query.test.ts:83]: Yep, that's it.

**Review by john-u — APPROVED** [review]

**Pull request merged**